tests/model_fields/test_generatedfield.py

```diff
@@ -93,7 +92,7 @@ def _refresh_if_needed(self, m):
     def test_unsaved_error(self):
         m = self.base_model(a=1, b=2)
         msg = "Cannot read a generated field from an unsaved model."
-        with self.assertRaisesMessage(FieldError, msg):
+        with self.assertRaisesMessage(AttributeError, msg):
```

Comment:
This is a side-effect of the changes rather than a proper regression test. Can you add a small test of admin view with a readonly generated field? (e.g. in `tests/modeladmin/tests.py`.)

Reply:
@felixxm I added a simple but I think effective regression test.
If the bug is reintroduced the test should raise an error